engine/src/test/java/io/zeebe/engine/state/variable/VariableStateTest.java

```diff
@@ -586,23 +589,24 @@ public void shouldNotInvokeListenerIfNotChanged() {
   @Test
   public void shouldInvokeListenerIfSetVariablesLocalFromDocument() {
     // given
+    final Map<String, Object> document = Map.of("x", "foo", "y", "bar");
 
     // when
-    setVariablesLocalFromDocument(parent, asMsgPack("{'x':'foo', 'y':'bar'}"));
+    setVariablesLocalFromDocument(parent, asMsgPack(document));
 
     // then
-    assertThat(listener.created).hasSize(2);
-    assertThat(listener.created.get(0).name).isEqualTo("x");
-    assertThat(listener.created.get(0).value).isEqualTo(stringToMsgpack("foo"));
-    assertThat(listener.created.get(0).variableScopeKey).isEqualTo(parent);
-    assertThat(listener.created.get(0).rootScopeKey).isEqualTo(parent);
-
-    assertThat(listener.created.get(1).name).isEqualTo("y");
-    assertThat(listener.created.get(1).value).isEqualTo(stringToMsgpack("bar"));
-    assertThat(listener.created.get(1).variableScopeKey).isEqualTo(parent);
-    assertThat(listener.created.get(0).rootScopeKey).isEqualTo(parent);
-
     assertThat(listener.updated).isEmpty();
+    assertThat(listener.created).hasSize(document.size());
```

Comment:
This test had to be changed as now the order isn't well defined, since we store the offsets in a map which is unordered. I don't particularly like how it's asserted though, if you know of an easier way I'm all ears :+1: 

At the same time, this whole listener stuff will be removed at the end so maybe it's not worth spending too much time on this.

Reply:
I would not spend time on it, if we're going to remove that listener construct anyways. Having said that, I would probably assert this by extracting each of the different parts into a tuple and verifying that it `containsExactlyInAnyOrder` the same tuples build from the document.entryset